### ISSUE TITLE
Sigmoid Focal Loss vs Softmax Focal Loss - Object Detection

### ISSUE THREAD

**Issue by shreyasc-13** [facebookresearch/detectron2#1779]
## 🚀 Feature
Replacing sigmoid focal loss (for binary classification) used by loss_cls in Retinanet for object detection with a softmax focal loss (the multi-class variant).

## Motivation & Examples

For object detection in Retinanet, the loss_cls is currently implemented as a sigmoid focal loss (https://github.com/facebookresearch/fvcore/blob/master/fvcore/nn/focal_loss.py). But this formulation only should hold for a binary classification problem. Since this is a multi-class classification problem, I believe the multi-class variant of focal loss - taking the softmax and using CrossEntropyLoss - should be used.

That is:
- https://github.com/facebookresearch/fvcore/blob/master/fvcore/nn/focal_loss.py#L33 is replaced by ```torch.nn.Softmax(inputs)```, and
- https://github.com/facebookresearch/fvcore/blob/master/fvcore/nn/focal_loss.py#L34 is replaced by ```F.cross_entropy(inputs, targets, reduction="none")```

If not, is there any reason for using the sigmoid focal loss  form currently being used?
Thanks.

**Comment by ppwwyyxx** [comment]
We use sigmoid focal loss in RetinaNet because the RetinaNet paper uses it.
New ideas can be tried or implemented separately, but we by default choose to follow the paper unless there is a compelling reason not to do so.
